lib/sidekiq/instrument/worker.rb

```diff
@@ -29,17 +29,21 @@ def perform
       Statter.statsd.gauge('shared.sidekiq.stats.working', working)
       Statter.dogstatsd&.gauge('sidekiq.working', working)
 
-      info.queues.each do |name, size|
-        Statter.statsd.gauge("shared.sidekiq.#{name}.size", size)
-        Statter.dogstatsd&.gauge('sidekiq.queue.size', size, tags: ["queue:#{name}"])
-      end
-
       Sidekiq::Queue.all.each do |queue|
+        Statter.statsd.gauge("shared.sidekiq.#{queue.name}.size", queue.size)
+        Statter.dogstatsd&.gauge('sidekiq.queue.size', queue.size, tags: dd_tags(queue))
+
         Statter.statsd.gauge("shared.sidekiq.#{queue.name}.latency", queue.latency)
-        Statter.dogstatsd&.gauge('sidekiq.queue.latency', queue.latency, tags: ["queue:#{queue.name}"])
+        Statter.dogstatsd&.gauge('sidekiq.queue.latency', queue.latency, tags: dd_tags(queue))
       end
 
       Statter.dogstatsd&.flush(sync: true)
     end
+
+    private
+
+    def dd_tags(queue)
+      { queue: queue.name }
```

Comment:
Is it okay that this is changing from an array to a map?

Reply:
Looks like `gauge` wants an array for the `tags` param: https://www.rubydoc.info/github/DataDog/dogstatsd-ruby/Datadog%2FStatsd:gauge#:~:text=%3Atags%20(Array%3CString%3E)%20%E2%80%94%20An%20array%20of%20tags